Flaky test failure on FileOperations.newLockingOutputStream

Got an interesting flaky test failure.

```
com.google.cloud.tools.jib.filesystem.FileOperationsTest > testNewLockingOutputStream_newFile FAILED
    java.lang.AssertionError: Lock attempt should have failed
        at org.junit.Assert.fail(Assert.java:88)
        at com.google.cloud.tools.jib.filesystem.FileOperationsTest.verifyWriteWithLock(FileOperationsTest.java:49)
        at com.google.cloud.tools.jib.filesystem.FileOperationsTest.testNewLockingOutputStream_newFile(FileOperationsTest.java:98)
```

Full log: https://source.cloud.google.com/results/invocations/c9d1aec1-105e-4225-89e4-129a3c80390a/log

Test code:
```java
  private static void verifyWriteWithLock(Path file) throws IOException {
    OutputStream fileOutputStream = FileOperations.newLockingOutputStream(file);

    // Checks that the file was locked.
    try {
      FileChannel.open(file, StandardOpenOption.READ).tryLock(0, Long.MAX_VALUE, true);
      Assert.fail("Lock attempt should have failed");

    } catch (OverlappingFileLockException ex) {
      // pass
    }
```

--- a/jib-core/src/test/java/com/google/cloud/tools/jib/filesystem/FileOperationsTest.java
+++ b/jib-core/src/test/java/com/google/cloud/tools/jib/filesystem/FileOperationsTest.java
@@ -26,6 +26,7 @@ import java.io.OutputStream;
 import java.net.URISyntaxException;
 import java.nio.channels.Channels;
 import java.nio.channels.FileChannel;
+import java.nio.channels.FileLock;
 import java.nio.channels.OverlappingFileLockException;
 import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
@@ -44,9 +45,10 @@ public class FileOperationsTest {
     OutputStream fileOutputStream = FileOperations.newLockingOutputStream(file);
 
     // Checks that the file was locked.
-    try {
-      FileChannel.open(file, StandardOpenOption.READ).tryLock(0, Long.MAX_VALUE, true);
-      Assert.fail("Lock attempt should have failed");
+    try (FileChannel channel = FileChannel.open(file, StandardOpenOption.READ)) {
+      // locking should either fail and return null or throw an OverlappingFileLockException
+      FileLock lock = channel.tryLock(0, Long.MAX_VALUE, true);
+      Assert.assertNull("Lock attempt should have failed", lock);
 
     } catch (OverlappingFileLockException ex) {
       // pass

